EchoPro/survey.py

```diff
@@ -944,10 +840,287 @@ def strata_age_binned_weight_proportions( self ,
+            .pipe( lambda df: df.assign( interval = df[ 'vessel_log_start' ].diff( periods = -1 ).abs( ) ) 
+                                .replace( np.nan , df[ 'vessel_log_end' ].iloc[ -1 ] - df[ 'vessel_log_start' ].iloc[ -1 ] ) )
+            # Replace likely erroneous interval lengths associated with the edges of each transect
+            .pipe( lambda df: df.assign( median_interval = np.median( df[ 'interval' ] ) )
+                                .assign( interval = lambda x: np.where( np.abs( x[ 'interval' ] - x[ 'median_interval' ] > 0.05 ) ,
+                                                                        x.vessel_log_end - x.vessel_log_start ,
+                                                                        x.interval ) ) )
+            # Calculate interval area
+            .pipe( lambda df: df.assign( interval_area = df[ 'interval' ] * df[ 'transect_spacing' ] ) )                            
+            # Keep dataframe tidy by only retaining the necessary columns/variables
+            .loc[ : , [ 'latitude' , 'longitude' , 'transect_num' , 'stratum_num' , 'haul_num' , 'interval_area' , 'NASC_all_ages' , 'NASC_no_age1' ] ]
+        )
+
+        ### Call additional dataframes needed to merge with the NASC data and subsequently calculate
+        ### population-level metrics (and later statistics)
+        # Sex-stratum-indexed proportions and average weight
+        weight_sex_strata = (
+            self.biology[ 'weight' ][ 'weight_strata_df' ]
+            .copy()
+            .rename( columns = { 'stratum': 'stratum_num' } )
+        )
+
+        # Stratum information
+        info_strata = self.spatial[ 'strata_df' ].copy()
+
+        # Stratum-averaged sigma_bs
+        sigma_bs_strata = self.acoustics[ 'sigma_bs' ][ 'strata_mean' ]
+
+        # Adult NASC proportions for each stratum (number)
+        # !!! TODO: Currently only uses 'age_1_excluded' -- this should become an argument that toggles
+        ## This is not a major issue since both 'NASC_*' columns can be pivoted to create a single 
+        ## NASC column so the column name does not have to be hard-coded. This could then correspond to
+        ## the configuration settings in some way, or this may be where the argument comes into play where
+        ## the dataframe can be simply filtered based on the input/selection.
+        # between excluding and including age-1 fish
+        nasc_adult_number_proportions = (
+            self.biology[ 'weight' ][ 'age_stratified' ][ 'age_1_excluded' ][ 'number_proportions' ]
+            .rename( columns = { 'number_proportion': 'adult_number_proportion' } )
+        )
+
+        # Adult NASC proportions for each stratum (weight)
+        nasc_adult_weight_proportions = (
+            self.biology[ 'weight' ][ 'age_stratified' ][ 'age_1_excluded' ][ 'weight_proportions' ]
+            .rename( columns = { 'weight_proportion': 'adult_weight_proportion' } )
+        )
```

Comment:
Same as the above, wondering if this renaming means that we could rename this column in the data loader.

Reply:
In these examples, I may need to rethink how I am originally generating the dataframes when calculating the sex/age proportions, or just not rename them at all in this code block. In my mind, I assume that someone accessing these age-specific trees understand that the associated `weight_proportion` and `number_proportion` are subsequently adult-only variables. I renamed the columns here since this function could be applied to either all age-classes, just YOY, or something else entirely. I'll take a step back and walk through it again to see where I can trim things, but I think your suggestion regarding further refactoring of the `nasc_to_biomass` function may be the simplest solution here.